### PR TITLE
[AMDGPU] - Add clang builtins for tied WMMA intrinsics

### PR DESCRIPTION
**Pull request by OutOfCache**
Add clang builtins for the new tied wmma intrinsics. 
These variations tie the destination
accumulator matrix to the input
accumulator matrix.

See https://github.com/llvm/llvm-project/pull/69903 for context.

### CODE REVIEW

**Review by rampitec — COMMENTED** [review]
Also needed negative tests that gfx11-insts feature is required (using gfx1030 target for example) and for the immediate arguments. See for example builtins-amdgcn-gfx11-err.cl and builtins-amdgcn-fp-atomics-gfx11-err.cl.

**Comment by nhaehnle** [comment]
(removing myself from this PR)

File: clang/include/clang/Basic/BuiltinsAMDGPU.def [hunk 1]
```diff
@@ -292,13 +292,17 @@ TARGET_BUILTIN(__builtin_amdgcn_wmma_f32_16x16x16_f16_w32, "V8fV16hV16hV8f", "nc
 TARGET_BUILTIN(__builtin_amdgcn_wmma_f32_16x16x16_bf16_w32, "V8fV16sV16sV8f", "nc", "gfx11-insts")
 TARGET_BUILTIN(__builtin_amdgcn_wmma_f16_16x16x16_f16_w32, "V16hV16hV16hV16hIb", "nc", "gfx11-insts")
 TARGET_BUILTIN(__builtin_amdgcn_wmma_bf16_16x16x16_bf16_w32, "V16sV16sV16sV16sIb", "nc", "gfx11-insts")
+TARGET_BUILTIN(__builtin_amdgcn_wmma_f16_16x16x16_f16_tied_w32, "V16hV16hV16hV16hIb", "nc", "gfx11-insts")
```
Comment (rampitec) [inline clang/include/clang/Basic/BuiltinsAMDGPU.def:295]: Need to add negative test for the last operand to always be a constant integer. We do it every time 'I' modifier is used.

Comment (OutOfCache) [reply clang/include/clang/Basic/BuiltinsAMDGPU.def:295]: I see, thanks for letting me know! I tried to do so now.

**Review by rampitec — APPROVED** [review]
LGTM

**Pull request merged**